heft-plugins/heft-sass-plugin/src/SassPlugin.ts

```diff
@@ -43,84 +45,125 @@ export default class SassPlugin implements IHeftPlugin {
+    // Otherwise, process all files. The typings generator also provides the file paths
+    // as relative paths from the sourceFolderPath.
+    let changedFilePaths: string[] | undefined;
+    if (runIncrementalOptions) {
+      changedFilePaths = [];
+      const filePaths: string[] = runIncrementalOptions.globChangedFiles(sassProcessor.inputFileGlob, {
+        cwd: sassProcessor.sourceFolderPath,
+        ignore: Array.from(sassProcessor.ignoredFileGlobs)
+      });
+      const deleteFilePromises: Promise<void>[] = [];
+      for (const filePath of filePaths) {
+        // Filter out and delete any files that are removed and all their output files
+        const absoluteFilePath: string = path.join(sassProcessor.sourceFolderPath, filePath);
+        if (runIncrementalOptions.changedFiles.get(absoluteFilePath)!.version === undefined) {
+          deleteFilePromises.push(
+            ...sassProcessor.getOutputFilePaths(filePath).map(async (outputFilePath) => {
+              await FileSystem.deleteFileAsync(outputFilePath);
+            })
+          );
+        } else {
+          changedFilePaths.push(filePath);
+        }
+      }
+      await Promise.all(deleteFilePromises);
+    }
+
+    await sassProcessor.generateTypingsAsync(changedFilePaths);
+    taskSession.logger.terminal.writeLine('Generated sass typings');
+  }
+
+  private async _loadSassProcessorAsync(
+    heftConfiguration: HeftConfiguration,
+    logger: IScopedLogger
+  ): Promise<SassProcessor> {
+    if (!this._sassProcessor) {
+      const sassConfiguration: ISassConfiguration = await this._loadSassConfigurationAsync(
+        heftConfiguration,
+        logger
       );
+      this._sassProcessor = new SassProcessor({ sassConfiguration });
     }
+    return this._sassProcessor;
   }
 
   private async _loadSassConfigurationAsync(
     heftConfiguration: HeftConfiguration,
     logger: IScopedLogger
   ): Promise<ISassConfiguration> {
-    const { buildFolderPath } = heftConfiguration;
-    const sassConfigurationJson: ISassConfigurationJson | undefined =
-      await SassPlugin._getSassConfigurationLoader().tryLoadConfigurationFileForProjectAsync(
-        logger.terminal,
-        buildFolderPath,
-        heftConfiguration.rigConfig
-      );
-
-    if (sassConfigurationJson) {
-      if (sassConfigurationJson.srcFolder) {
-        sassConfigurationJson.srcFolder = path.resolve(buildFolderPath, sassConfigurationJson.srcFolder);
-      }
-
-      if (sassConfigurationJson.generatedTsFolder) {
-        sassConfigurationJson.generatedTsFolder = path.resolve(
+    if (!this._sassConfiguration) {
+      const { buildFolderPath } = heftConfiguration;
+      const configurationLoader: ConfigurationFile<ISassConfigurationJson> =
+        new ConfigurationFile<ISassConfigurationJson>({
+          projectRelativeFilePath: SASS_CONFIGURATION_LOCATION,
+          jsonSchemaPath: PLUGIN_SCHEMA_PATH
+        });
```

Comment:
If for some reason the sass plugin was present in multiple phases, this will duplicate the configuration object. Same would occur if we add support for heft server mode and allow reuse across projects. There's a reason for the static loader instance.

Reply:
Will do it for the loader, but not going to do it for the loaded config object. That will be something that becomes an issue if/when we move to using Heft as a service.